test/qctrl/test_qctrl.py

```diff
@@ -0,0 +1,394 @@
+        self.assertGreater(fidelity, FIDELITY_THRESHOLD)
+
+    @run_integration_test
+    def test_sampler_qctrl_superposition(self, service):
+        """Test qctrl small superposition"""
+
+        shots = 1000
+        num_qubits = 5
+        superposition_circuit = QuantumCircuit(num_qubits)
+        superposition_circuit.h(range(num_qubits))
+
+        # Add measurements for the sampler
+        superposition_circuit_sampler = superposition_circuit.copy()
+        superposition_circuit_sampler.measure_active()
+
+        # Execute circuit in a session with sampler
+        with Session(service, backend=self.backend):
+            options = Options(resilience_level=1)
+            sampler = Sampler(options=options)
+
+            result = sampler.run(superposition_circuit_sampler, shots=shots).result()
+            results_dict = {
+                f"{{0:0{num_qubits}b}}".format(key): value
+                for key, value in result.quasi_dists[0].items()
+            }  # convert keys to bitstrings
+
+        ideal_result = {
+            key: val / shots
+            for key, val in Statevector(superposition_circuit).probabilities_dict().items()
+        }
+        fidelity = hellinger_fidelity(results_dict, ideal_result)
+        self.assertGreater(fidelity, FIDELITY_THRESHOLD)
+
+    @run_integration_test
+    def test_sampler_qctrl_conditional_states(self, service):
+        """Test qctrl conditional states"""
+        shots = 1000
+        num_qubits = 3
+        computational_states_circuits = []
+        for idx in range(2**num_qubits):
+            circuit = QuantumCircuit(num_qubits)
+            bitstring = f"{{0:0{num_qubits}b}}".format(idx)
+            for bit_pos, bit in enumerate(
+                bitstring[::-1]
+            ):  # convert to little-endian (qiskit convention)
+                if bit == "1":
+                    circuit.x(bit_pos)
+            computational_states_circuits.append(circuit)
+
+        # Add measurements for the sampler
+        computational_states_sampler_circuits = []
+        for circuit in computational_states_circuits:
+            circuit_sampler = circuit.copy()
+            circuit_sampler.measure_all()
+            computational_states_sampler_circuits.append(circuit_sampler)
+
+        # Execute circuit in a session with sampler
+        with Session(service, backend=self.backend):
+            options = Options(resilience_level=1)
+            sampler = Sampler(options=options)
+
+            result = sampler.run(computational_states_sampler_circuits, shots=shots).result()
+            results_dict_list = [
+                {f"{{0:0{num_qubits}b}}".format(key): value for key, value in quasis.items()}
+                for quasis in result.quasi_dists
+            ]  # convert keys to bitstrings
+
+        ideal_results_list = [
+            {key: val / shots for key, val in Statevector(circuit).probabilities_dict().items()}
```

Comment:
Seems like an expensive way to compute something which you already know: the target bitstring should be '1' and everything else a '0'. But, okay...